File: bench/benches/primes.py
import sys, os
sys.path.insert(0, os.path.join(os.path.dirname(__file__), "..", "lib"))
from bench import bench

def is_prime(n):
    d = 2
    while d * d <= n:
        if n % d == 0:
            return False
        d += 1
    return True

def count(lo, hi):
    c = 0
    for n in range(lo, hi):
        if is_prime(n):
            c += 1
    return c

bench("primes", lambda: count(2, 30000))

File: bench/benches/primes.rs
include!("../lib/bench.rs");

fn is_prime(n: i64) -> bool {
    let mut d = 2;
    while d * d <= n {
        if n % d == 0 {
            return false;
        }
        d += 1;
    }
    true
}

fn count(lo: i64, hi: i64) -> i64 {
    let mut c = 0;
    for n in lo..hi {
        if is_prime(n) {
            c += 1;
        }
    }
    c
}

fn main() {
    bench("primes", || count(2, 30000));
}
